### PR TITLE
Add a tactic which uses z3 to prove goals

### PR DESCRIPTION
**Pull request by jroesch**
This is an initial tactic which uses z3 to prove a subset of Lean goals, and uses sorry to finish the proof. 

Currently it supports a limited subset of Lean, it will translate uninterpreted functions, propositional logic & arithmetic into a z3 query.

Doing a clean-up pass right now, but we should merge the initial version. Since we are lacking arithmetic support in Lean currently this might be generally useful.

### CODE REVIEW

File: tests/lean/run/smt2_P_implies_Q_fail.lean [hunk 1]
```diff
@@ -0,0 +1,5 @@
+example (P Q : Prop) : P -> Q :=
+begin
+    intros,
+    z3
```
Comment (gebner) [inline tests/lean/run/smt2_P_implies_Q_fail.lean:4]: This introduces a hard build dependency on Z3 in order to run the tests.  I think we should move the Z3 tests into a separate directory and only run them if we can find the `z3` executable.

File: tests/lean/run/smt2_p_implies_p.lean [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import tools.smt2
+
+example (P : Prop) : P → P :=
+begin
+    intros,
+    z3 $ "/tmp/P.smt2"
```
Comment (gebner) [inline tests/lean/run/smt2_p_implies_p.lean:6]: Does this work on windows?

Comment (jroesch) [reply tests/lean/run/smt2_p_implies_p.lean:6]: This was leftover from testing, got distracted yesterday afternoon by other things and haven't had a chance to come back yet.

File: tests/lean/run/smt2_P_implies_Q_fail.lean (hunk 1 above)
Comment (jroesch) [reply tests/lean/run/smt2_P_implies_Q_fail.lean:4]: I agree, hadn't figured out the right place to put them yet. Maybe we should just move this code out of the standard library into a package? I don't know how far along the package management tooling is.

**Comment by leodemoura** [comment]
@jroesch Are you going to fix this PR? Or, should I close it?
It cannot be merged as is because of the problems raised by @gebner.

**Comment by leodemoura** [comment]
This PR was merged into a separate repo: https://github.com/leanprover/smt2_interface

**Pull request closed without merging**